src/python/pants/goal/run_tracker.py

```diff
@@ -414,21 +414,77 @@ def shutdown_worker_pool(self):
     """
     SubprocPool.shutdown(self._aborted)
 
-  def report_target_info(self, scope, target, key, val):
+  @staticmethod
+  def create_dict_with_nested_keys_and_val(all_keys, value, index):
+    """Recursively constructs a nested dictionary with the keys pointing to the value.
+
+    :param list of string all_keys: A list of keys to be nested as a dictionary.
+    :param dict or string value: The value of the information being stored.
+    :param int index: The index of the last element of the list of keys.
```

Comment:
Instead of passing in a list and an index, why not pass in a truncated copy of the list? The code will be more readable, and the performance implications are almost certainly unmeasurable.